sunpy/map/mapbase.py

```diff
@@ -1941,19 +1941,16 @@ def draw_rectangle(self, bottom_left, *, width: u.deg = None, height: u.deg = No
         Extra keyword arguments to this function are passed through to the
         `~matplotlib.patches.Rectangle` instance.
         """
-        if isinstance(top_right, u.Quantity) and isinstance(width, u.Quantity):
-            # The decorator assigns the first positional arg to top_right and so on.
-            height = width
-            width = top_right
-            top_right = None
-
         bottom_left, top_right = get_rectangle_coordinates(bottom_left,
                                                            top_right=top_right,
                                                            width=width,
                                                            height=height)
 
+        bottom_left = bottom_left.transform_to(self.coordinate_frame)
+        top_right = top_right.transform_to(self.coordinate_frame)
```

Comment:
This is a major behaviour change if your coordinate frame of the corners and map is different right? Even if it's a bug fix the specific change in behavour should be clearly documented in the changelog.

Reply:
Well, it's a major behavior change in the sense that it's a fix of a major bug.  Previously, `bottom_left` was transformed to the map's coordinate frame only *after* the width and height of the rectangle had already been calculated, but those width and height values are used as if they are in the map's coordinate frame (e.g., a rectangle specified with a width of 10 degrees in HGS would be plotted as an rectangle with a width of 10 degrees in HPC).  This PR fixes the logic.

Reply:
I've updated the changelog